staging/src/k8s.io/apiserver/pkg/util/flowcontrol/reqmgmt-config.go

```diff
@@ -0,0 +1,350 @@
+		return false
+	}
+	newFSs, err := reqMgr.fsLister.List(all)
+	if err != nil {
+		klog.Errorf("Unable to list FlowSchema objects: %s", err.Error())
+		return false
+	}
+	reqMgr.digestConfigObjects(newPLs, newFSs)
+	return true
+}
+
+func (reqMgr *requestManager) digestConfigObjects(newPLs []*rmtypesv1a1.PriorityLevelConfiguration, newFSs []*rmtypesv1a1.FlowSchema) {
+	oldRMState := reqMgr.curState.Load().(*requestManagerState)
+	var shareSum float64
+	newRMState := &requestManagerState{
+		priorityLevelStates: make(map[string]*priorityLevelState),
+	}
+	newlyQuiescent := make([]*priorityLevelState, 0)
+	var haveExemptPL, haveCatchAllPL, haveExemptFS, haveCatchAllFS bool
+	for _, pl := range newPLs {
+		state := oldRMState.priorityLevelStates[pl.Name]
+		if state == nil {
+			state = &priorityLevelState{
+				config: pl.Spec,
+			}
+		} else {
+			oState := *state
+			state = &oState
+			state.config = pl.Spec
+			if state.emptyHandler != nil { // it was undesired, but no longer
+				klog.V(3).Infof("Priority level %q was undesired and has become desired again", pl.Name)
+				state.emptyHandler = nil
+				state.queues.Quiesce(nil)
+			}
+		}
+		if state.config.Limited != nil {
+			shareSum += float64(state.config.Limited.AssuredConcurrencyShares)
+			qsConfig, err := qscOfPL(pl, reqMgr.requestWaitLimit)
+			if err != nil {
+				klog.Warningf(err.Error())
+				continue
+			}
+			state.qsConfig = qsConfig
+		} else {
+			haveExemptPL = true
+		}
+		haveCatchAllPL = haveCatchAllPL || pl.Name == rmtypesv1a1.PriorityLevelConfigurationNameCatchAll
+		newRMState.priorityLevelStates[pl.Name] = state
+	}
+
+	fsSeq := make(apihelpers.FlowSchemaSequence, 0, len(newFSs))
+	for i, fs := range newFSs {
+		_, flowSchemaExists := newRMState.priorityLevelStates[fs.Spec.PriorityLevelConfiguration.Name]
+		reqMgr.syncFlowSchemaStatus(fs, !flowSchemaExists)
+		if flowSchemaExists {
+			fsSeq = append(fsSeq, newFSs[i])
+			haveExemptFS = haveExemptFS || fs.Name == rmtypesv1a1.FlowSchemaNameExempt
+			haveCatchAllFS = haveCatchAllFS || fs.Name == rmtypesv1a1.FlowSchemaNameCatchAll
+		}
+	}
+	sort.Sort(fsSeq)
+
+	for plName, plState := range oldRMState.priorityLevelStates {
+		if newRMState.priorityLevelStates[plName] != nil {
+			// Still desired and already updated
+			continue
+		}
+		newState := *plState
+		plState = &newState
+		if plState.emptyHandler != nil && plState.emptyHandler.IsEmpty() {
+			// The queues are empty and will never get any more requests
+			plState.queues = nil
+			plState.emptyHandler = nil
+			klog.V(3).Infof("Retired queues for undesired quiescing priority level %q", plName)
+		}
+		if plState.config.Limited == nil && !haveExemptPL || plName == rmtypesv1a1.PriorityLevelConfigurationNameCatchAll && !haveCatchAllPL {
+			klog.V(3).Infof("Retaining old priority level %q with Type=%v because of lack of replacement", plName, plState.config.Type)
+		} else {
+			if plState.queues == nil { // should never happen; but if it does:
```

Comment:
The "should never happen" comment here is wrong, this _can_ happen.  In fact, this is the normal way to get a priority level out of the data structure.

Reply:
Bogus comment removed.